File: 0017-letter-combinations-of-a-phone-number/solution.go
func letterCombinations(digits string) []string {
    if digits == "" {
        return []string{}
    }

    dial := []string{"abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"}
    var result []string

    var backtrack func(curr string, number string)
    backtrack = func(curr string, number string) {
        if number == "" {
            result = append(result, curr)
        } else {
            chars := dial[number[0] - '2']
            for _, char := range chars {
                backtrack(curr + string(char), number[1:])
            }
        }
    }

    backtrack("", digits)
    return result
}

File: 0017-letter-combinations-of-a-phone-number/solution.py
import itertools

class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        if not digits:
            return []
        
        dtl = {
                "2": "abc", "3": "def", "4": "ghi", "5": "jkl", 
                "6": "mno", "7": "pqrs", "8": "tuv", "9": "wxyz"}
        
        ans = list(itertools.product(*[dtl[d] for d in digits]))
        
        return [''.join(lst) for lst in ans]